Add overflow event for non scrolling consoles

SlySven opened the pull request:
#### Brief overview of PR changes/additions
When a non-scrolling `TConsole` has text added to it that exceeds the number of lines it can show this raises a `sysWindowOverflowEvent` with the name of the console and the number of lines that have "scrolled off the top" i.e. that have been hidden because there is not enough space for them all. This can be used to manage the size of the text kept in that window so that it is always fully visible.

#### Motivation for adding to Mudlet
This was prompted by the second part of my comment:
https://github.com/Mudlet/Mudlet/pull/6848#pullrequestreview-1425398702

#### Other issues
As I was looking at code that examines the number of lines of text in the buffer I spotted that there were a couple of cases where the size of the buffer or the size of a line in that buffer was being taken and then being compared to be `<1` when really an `isEmpty()` test is a clearer indication of intent - and where assignment to a temporary `(int) length` variable could be made more streamlined.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 committed:
Infrastructure: add open bounties to readme (#6833)

<!-- Keep the title short & concise so anyone non-technical can
understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Add open bounties to readme
#### Motivation for adding to Mudlet
More visibility?
#### Other info (issues closed, discussion etc)

vadi2 committed:
Infrastructure: switch display performance builds to use cmake (#6832)

<!-- Keep the title short & concise so anyone non-technical can
understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Switch performance builds over to cmake
#### Motivation for adding to Mudlet
QMake is going away, future resiliency
#### Other info (issues closed, discussion etc)

vadi2 committed:
Infrastructure: update contributing guide for html escaping (#6831)

<!-- Keep the title short & concise so anyone non-technical can
understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Need to escape dynamic information in labels otherwise html strings in
trigger/alias/etc names will mess the display up, see
https://github.com/Mudlet/Mudlet/pull/6807
#### Motivation for adding to Mudlet
So `<`'s in item names work fine, mainly
#### Other info (issues closed, discussion etc)
Could not think of a better name than "dynamic information"

SlySven committed:
Add: overflow event for non-scrolling TConsoles

This was prompted by the second part of my comment:
https://github.com/Mudlet/Mudlet/pull/6848#pullrequestreview-1425398702

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Merge branch 'development' into add_overflowEventForNonScrollingConsoles

vadi2 commented on src/TConsole.cpp:
```diff
@@ -2348,3 +2348,33 @@ void TConsole::slot_clearSearchResults()
     mLowerPane->forceUpdate();
 }
+
+void TConsole::checkNumberOfLines(const int lineCount)
```
I think this is hitting https://www.linguistic-antipatterns.com/?tab=%22Unexpected-side-effects%22, it says "check" but it doesn't just check but also raises an event. 

Just throwing an idea out there so I just don't have critique only, how about something like this:

if (exceededConsoleHeight(lineCount)) {
  raiseOverflowEvent();
}

SlySven replied:
I was trying to avoid something that was too long `checkAndRaiseEventOnExcessLinesInNonScrollingConsole(...)`!

As you can see it tries to find a reason ***not*** to raise the event by checking progressively more "involved" things until it has eliminated them all...

SlySven replied:
How about `raiseLinesOverflowEventIfRequired(...)`

vadi2 replied:
Hmm I was thinking of splitting up the checking and raising also for a separation of concerns reason... but if we're sticking with one function, how about something more concise like `handleLinesOverflowEvent()`?

SlySven replied:
Okay I'll change it (again) to that...

eowmob committed:
Improvement: MXP SEND menus can be extended with MXP entities (#6852)

<!-- Keep the title short & concise so anyone non-technical can
understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
MXP entities in send tags are interpolated before HREF and HINT lists
are split into part
#### Motivation for adding to Mudlet
Enhance MXP compatibility with other clients (like mushclient and cmud)
and MXP using muds.
#### Other info (issues closed, discussion etc)
To see the issue / change in action:

connect to aldebaran-mud.de port 2000
login as guest
y     (confirm you want to login as a guest)

Notice the mouse-over hint on the ... elipsis in the default prompt.
Right click and try some commands.

Now do:
alias bh ^BE HAPPY^bounce hap

Try entering 'bh' if you like. Do the mouse over again (in one of the
prompts printed after the alias cmd), right click
and try the "BE HAPPY" command from the menu.

SlySven committed:
Merge branch 'development' into add_overflowEventForNonScrollingConsoles

SlySven committed:
Revise: rename a method to prevent a naming anti-pattern

This was suggested by @vadi2 in:
https://github.com/Mudlet/Mudlet/pull/6872#discussion_r1208635363

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: change the method name again!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
Sheesh! I thought I was setting it up to auto-merge...! :scream:
